### PR TITLE
Add oshinko sparklyr S2i image

### PR DESCRIPTION
**Pull request by rimolive**

### CODE REVIEW

**Comment by willb** [comment]
@sophwats can you review this?

File: image.sparklyr.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: "radanalyticsio/radanalytics-sparklyr"
+version: 1.0
+release: dev
+from: "rimolive/r-34-centos7:latest"
```
Comment (tmckayus) [inline image.sparklyr.yaml]: for the moment, this is okay. But we should probably add the base image to docker.io/radanalyticsio

File: image.sparklyr.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  - name: "io.k8s.display-name"
+    value: "radanalytics sparklyr"
+  - name: "io.openshift.expose-services"
+    value: "8080:http"
+  - name: "io.openshift.s2i.scripts-url"
+    value: "image:///usr/libexec/s2i"
+  - name: "io.openshift.tags"
+    value: "builder,radanalytics,sparklyr"
+  - name: "io.radanalytics.sparkdistro"
+    value: "https://archive.apache.org/dist/spark/spark-2.2.0/spark-2.2.0-bin-hadoop2.7.tgz"
```
Comment (tmckayus) [inline image.sparklyr.yaml]: the other s2i images are currently at 2.2.1, and will move to 2.3.0 in the next release

**Comment by tmckayus** [comment]
Looks like this should be moved to concreate.  The other languages have moved, and there are no conflicts in  this PR since the scripts directory has been replaced by "modules".

However, the basics can still be tested using a dogen build.

File: image.sparklyr.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  - /usr/local/s2i/usage
+scripts:
+  - package: common
+    exec: install
+  - package: sparklyr
+    exec: install
+sources:
+  - url: https://archive.apache.org/dist/spark/spark-2.2.0/spark-2.2.0-bin-hadoop2.7.tgz
+    md5sum: 1715b661bdf33b40c98b3daa7837f690
+  - url: https://github.com/radanalyticsio/oshinko-cli/releases/download/v0.4.3/oshinko_v0.4.3_linux_amd64.tar.gz
```
Comment (tmckayus) [inline image.sparklyr.yaml]: this should be moved to the current oshinko release as well

File: scripts/sparklyr/added/s2i/usage [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+** Important! ** The APP_FILE environment variable specifies the name of the python file that
+contains the main routine for your pyspark application. The default value is
+"app.py". If your main routine is in a different file, you must use the APP_FILE
+option to s2i build to specify it.
+
+Example:
+
+    $s2i build -e APP_FILE=my_main.py git://github.com/mygithub/wordcount.git radanalytics-pyspark mynewapp
+
```
Comment (sophwats) [inline scripts/sparklyr/added/s2i/usage]: This needs to be updated to be relevant for the sparklyr s2i.

File: sparklyr-build/scripts/sparklyr/added/s2i/usage [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+** Important! ** The APP_FILE environment variable specifies the name of the python file that
+contains the main routine for your pyspark application. The default value is
+"app.py". If your main routine is in a different file, you must use the APP_FILE
+option to s2i build to specify it.
+
+Example:
+
+    $s2i build -e APP_FILE=my_main.py git://github.com/mygithub/wordcount.git radanalytics-pyspark mynewapp
+
```
Comment (sophwats) [inline sparklyr-build/scripts/sparklyr/added/s2i/usage]: Again needs updating to info for sparklyr S2i image

File: test/e2e/resources/scalabuilddc.json [hunk 6]
```diff
@@ -0,0 +1,357 @@
+{
```
Comment (tmckayus) [inline test/e2e/resources/scalabuilddc.json]: This file shouldn't be added, it's left over from a test run

File: test/e2e/resources/scalabuild.json [hunk 7]
```diff
@@ -0,0 +1,136 @@
+{
```
Comment (tmckayus) [inline test/e2e/resources/scalabuild.json]: This file shouldn't be added, it's left over from a test run

File: templates/scaladc.json [hunk 8]
```diff
@@ -1,101 +1,218 @@
 {
```
Comment (tmckayus) [inline templates/scaladc.json]: why did this file change? I don't think it should as part of this PR

File: Makefile [hunk 9]
```diff
@@ -85,6 +94,7 @@ test-e2e:
 	LOCAL_IMAGE=$(S2I_TEST_IMAGE_PYSPARK) make -f Makefile.pyspark build
 	LOCAL_IMAGE=$(S2I_TEST_IMAGE_JAVA) make -f Makefile.java build
 	LOCAL_IMAGE=$(S2I_TEST_IMAGE_SCALA) make -f Makefile.scala build
+	LOCAL_IMAGE=$(S2I_TEST_IMAGE_SPARKLYR) make -f Makefile.sparklyr build
 	test/e2e/run.sh
```
Comment (tmckayus) [inline Makefile]: hmm, just an aside, I bet this is supposed to be "test/e2e/run.sh operations".  Not part of this PR but I'll look.

**Pull request merged**